src/__tests__/dpsl.test.js

```diff
@@ -510,4 +530,47 @@ describe('dpsl.diagnostics tests', () => {
       });
     });
   });
+
+
+  test('runSmartctlCheckRoutine works without parameters', (done) => {
+    const expectedRunRoutineResponse = {id: 123456};
+
+    const chrome = {
+      os: {
+        diagnostics: {
+          runSmartctlCheckRoutine:
+             (params = undefined) =>
+               Promise.resolve(expectedRunRoutineResponse),
+        },
+      },
+    };
+    global.chrome = chrome;
+
+    dpsl.diagnostics.nvme.runSmartctlCheckRoutine().then((routine) => {
+      expect(routine).toEqual(expectedRunRoutineResponse);
+      done();
+    });
+  });
+
+  test('runSmartctlCheckRoutine works with parameters', (done) => {
+    const expectedRunRoutineResponse = {id: 123456};
+
+    const chrome = {
+      os: {
+        diagnostics: {
+          runSmartctlCheckRoutine:
+             (params = undefined) =>
+               Promise.resolve(params ? expectedRunRoutineResponse : undefined),
```

Comment:
Same here: we can use arguments to check that params was provided and we can validate that params. percentageUsedThreshold is equal to 42 before resolving the promise

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/arguments